crates/nostr/src/nips/nip22.rs

```diff
@@ -46,6 +46,75 @@ pub enum CommentTarget<'a> {
     },
 }
 
+impl<'a> CommentTarget<'a> {
+    /// Creates a new [`CommentTarget`] pointing to a specific event.
+    #[inline]
+    pub fn event(
+        event_id: &'a EventId,
+        kind: &'a Kind,
+        author: Option<&'a PublicKey>,
+        relay_hint: Option<&'a RelayUrl>,
+    ) -> Self {
+        Self::Event {
+            relay_hint,
+            id: event_id,
+            pubkey_hint: author,
+            kind: Some(kind),
+        }
+    }
+
+    /// Create a new [`CommentTarget`] pointing to a specific coordinate.
+    #[inline]
+    pub fn coordinate(coordinate: CoordinateBorrow<'a>, relay_hint: Option<&'a RelayUrl>) -> Self {
+        Self::Coordinate {
+            address: coordinate,
+            relay_hint,
+            kind: Some(coordinate.kind),
```

Comment:
Not sure why the CI don't return an error, here should be added `#[allow(deprecated)]`, on the `kind`